File: typescript-examples/network-interception/utils/typesAndSchemas.ts
import { z } from "zod";

export const paramsSchema = z.object({
  username: z.string(),
  password: z.string(),
  limit: z.number().default(5),
});

export const apiInterceptorParamsSchema = z.object({
  url: z.string(),
  api_pattern: z.string().default("/api/"),
  max_pages: z.number().default(10),
});

export const consultationSchema = z.object({
  id: z.string(),
  created_at: z.string(),
  name: z.string(),
  email: z.string(),
  phone: z.string(),
  preferred_date: z.string(),
  preferred_time: z.string(),
  topic: z.string(),
  status: z.string(),
  user_id: z.string().optional(),
});

File: python-examples/network-interception/utils/types_and_schemas.py
from pydantic import BaseModel, Field


class NetworkInterceptorParams(BaseModel):
    username: str = Field(..., description="Email used to log in to sandbox.intuned.dev")
    password: str = Field(..., description="Password used to log in to sandbox.intuned.dev")
    limit: int = Field(default=5, description="Maximum number of consultations to fetch")


class ApiInterceptorParams(BaseModel):
    url: str = Field(..., description="The URL to navigate to")
    api_pattern: str = Field(
        default="/api/",
        description="Pattern to match in API URLs (e.g., '/rest/v1/consultations')",
    )
    max_pages: int = Field(
        default=10, description="Maximum number of pages to fetch (default: 10)"
    )


class Consultation(BaseModel):
    id: str = Field(..., description="The ID of the consultation")
    created_at: str = Field(
        ..., description="The date and time the consultation was created"
    )
    name: str = Field(..., description="The name of the consultation")
    email: str = Field(..., description="The email of the consultation")
    phone: str = Field(..., description="The phone number of the consultation")
    preferred_date: str = Field(
        ..., description="The preferred date of the consultation"
    )
    preferred_time: str = Field(
        ..., description="The preferred time of the consultation"
    )
    topic: str = Field(..., description="The topic of the consultation")
    status: str = Field(..., description="The status of the consultation")
    user_id: str = Field(
        ..., description="The ID of the user who created the consultation"
    )
